backend/inspirehep/records/links.py

```diff
@@ -47,6 +47,15 @@ def inspire_search_links(links):
         .get("search_serializers_aliases", {})
         .keys()
     )
+    request_fields = request.values.get("fields", "", type=str)
+    if request_fields:
+        url_fields = f"&fields={request_fields}"
+        self_url += url_fields
+        links["self"] = self_url
+        next_url = links.get("next")
+        if next_url:
+            next_url += url_fields
+            links["next"] = next_url
```

Comment:
It should be only for `self` and `json`, for the other formats is useless AFAIR

Reply:
You are right, removed it for other formats. It should also be for ```next``` though.

Reply:
@monaawi and `previous`